rpcserver.go

```diff
@@ -6363,7 +6363,13 @@ func (r *rpcServer) ChannelAcceptor(stream lnrpc.Lightning_ChannelAcceptorServer
 	// Create a new RPCAcceptor which will send requests into the
 	// newRequests channel when it receives them.
 	rpcAcceptor := chanacceptor.NewRPCAcceptor(
-		stream.Recv, stream.Send, r.cfg.AcceptorTimeout, r.quit,
+		stream.Recv, stream.Send, r.cfg.AcceptorTimeout,
+		func(address string) (lnwire.DeliveryAddress, error) {
+			return parseUpfrontShutdownAddress(
```

Comment:
non-blocking:
Might be better to export this function and then pass in `ActiveNetParams` to the `RPCAcceptor`

Reply:
Problem here is that we can't import something in `rpcserver` into the `chanacceptor` package because we get a circular dep. Could move it and export, but I couldn't pick a logical place for it because it's used in a few places in the server.

Reply:
Maybe the `chancloser` package?